pbj-core/pbj-compiler/src/main/java/com/hedera/pbj/compiler/impl/generators/protobuf/CodecParseMethodGenerator.java

```diff
@@ -173,6 +173,9 @@ static String generateParseInternalMethod(final String modelClassName, final Lis
                         }
                         return new $modelClassName($fieldsList);
                     } catch (final Exception anyException) {
+                        if (anyException instanceof ParseException parseException) {
```

Comment:
Interesting solution. Why didn't you add a second catch-block?

Reply:
ParseException isn't always thrown (depends on the "case statements" produced below), only when there is a child message, so the compiler would fail in many cases with the catch block approach.